### DISCUSSION TITLE
Issue when adding a grid item

### DISCUSSION THREAD

**Discussion by Fyddilfo** [katoid/angular-grid-layout#45] [Q&A]
Hi !
On the playground, when you add a new item, the x and y values are not updated.
Let's say i have one bloc on my grid, this.layout will be : 
![image](https://user-images.githubusercontent.com/103193420/162201122-bbeda172-2a82-4a80-aa65-1b2ede649345.png)
when i add an item : 
![image](https://user-images.githubusercontent.com/103193420/162201204-36e75904-501a-47bb-bcb5-05da33097f31.png)

I have a draggable block that contains toggles to add/remove blocks, and when the user hit close, i save the layout. But, if the user didn't trigger the layoutUpdated event, the data will not be the correct one.
Am i missing something ?
thanks :)

**Comment by llorenspujol** [comment]
Hi!

Right now, adding and deleting elements in the grid are not handled by the grid itself. `layoutUpdated` would ONLY be called when you move or resize items in the grid. Internally, layout is well calculated because the Input property `compactOnPropsChange` is set to true.

What is the standard way to do it now, would be to call the compact function `ktdGridCompact` yourself after adding or removing and element in the layout array. This way, even the positions of the items are wrongly placed (as your example shows), `ktdGridCompact` would calculate a valid well-positioned layout.
I also would set the Input property `compactOnPropsChange` to false. (I may upgrade a little bit the docs usage, since this property is better being false)

This is the way to go right now for several reasons, if you have more doubts or this answer does not resolve your problem, give me an stackblitz example where I can see exactly what the issue is.

Hope this helps!
